public/script.js

```diff
@@ -0,0 +1,260 @@
+import { getCategories, getToken, getPlaylists, search } from './api.js';
+
+// html элемент, в котором находится основной контент на странице
+const content = document.querySelector('.content');
+// поисковая строка
+const input = document.querySelector('.header__search');
+
+/** 
+ * Функция для создания заготовки раздела страницы
+ * @returns html элемент section, состоящего из h2 - заголовка и div - контейнера для хранения элементов
+ */
+function createSection() {
+  const section = document.createElement('section');
+  section.className = 'section';
+  const template = `
+  <h2 class="section__header"></h2>
+  <div class="items"></div>
+  `;
+  section.insertAdjacentHTML('beforeend', template);
+  return section;
+}
+
+/**
+ *  Функция для создания элемента раздела страницы на основе переданного объекта 
+ * @returns ссылку, состоящую из изображения и подписи
+*/
+function objectToHtmlElement(obj) {
+  const a = document.createElement('a');
+  a.tabIndex = 0;
+  a.classList = 'item link footer__item-link';
+  a.target= '_blank';
+  a.href = obj.external_urls.spotify;
+  const image_url = obj.images.length == 0 ? "photos/default.jpg" : obj.images[0].url;
+  const template = `
+  <img class="item-image" src="${image_url}" alt="image" width="220" height="220">
+  <p class="item-header">${obj.name}</p>
+  `;
+  a.insertAdjacentHTML('beforeend', template);
+  return a;
+}
+
+/**
+*  Функция для создания раздела с плейлистами 
+* @param {object[]} playlists - массив плейлистов
+* @param {string} categoryName - название категории плейлистов
+* @returns раздел, содержащий название категории и ссылки на плейлисты из категории
+*/
+function playlistsToHtmlElement(playlists, categoryName) {
+  const playlistSection = createSection().cloneNode(true);
+  playlistSection.querySelector('.section__header').textContent = categoryName;
+  const fragment = document.createDocumentFragment();
+  playlists.forEach(playlist => {
+    const element = objectToHtmlElement(playlist);
+    element.insertAdjacentHTML('beforeend', `<p>${playlist.description}</p>`);
+    fragment.appendChild(element);
+  });
+  playlistSection.querySelector('.items').appendChild(fragment);
+  return playlistSection;
+}
+
+/**
+* Вспомогальная функция для преобразования массива с именами исполнителей в строку
+* @param artists - массив с именами исполнителей
+* @returns строку с именами исполнителей 
+*/
+function artistsToString(artists) {
+  let result = '';
+  artists.forEach(function(artist, index) {
```

Comment:
Можно сделать намного проще
Например так:
```suggestion
  return artists.map((artist)=>artist.name).join(', ')
```